UE/Application/Ports/BtsPort.cpp

```diff
@@ -1,146 +1,172 @@
-            logger.logDebug("BtsPort, SmsRecieved msg: ", message);
-            handler->handleSmsReceived(from, message);
-            break;
-        }
-        case common::MessageId::CallRequest:
-        {
-            logger.logDebug("BtsPort, Call request from:", from);
-            handler->handleCallRequest(from);
-            break;
-        }
-        case common::MessageId::CallAccepted:
-        {
-            logger.logDebug("BtsPort, Call accepted from", from);
-            handler->handleCallAccepted(from);
-            break;
-        }
-        case common::MessageId::CallTalk:
+    void BtsPort::handleMessage(BinaryMessage msg)
+    {
+        try
         {
-            logger.logDebug("BtsPort, Call talk", from);
-            handler->handleTalkMessage(reader.readRemainingText());
-            break;
+            common::IncomingMessage reader{msg};
+            auto messageId = reader.readMessageId();
+            common::PhoneNumber from = reader.readPhoneNumber();
+            auto to = reader.readPhoneNumber();
+
+            switch (messageId)
+            {
+            case common::MessageId::Sib:
+            {
+                auto btsId = reader.readBtsId();
+                handler->handleSib(btsId);
+                break;
+            }
+            case common::MessageId::AttachResponse:
+            {
+                bool accept = reader.readNumber<std::uint8_t>() != 0u;
+                if (accept)
+                    handler->handleAttachAccept();
+                else
+                    handler->handleAttachReject();
+                break;
+            }
+            case common::MessageId::Sms:
+            {
+                std::string message = reader.readRemainingText();
+                logger.logDebug("BtsPort, SmsRecieved from: ", from);
+                logger.logDebug("BtsPort, SmsRecieved msg: ", message);
+                handler->handleSmsReceived(from, message);
+                break;
+            }
+            case common::MessageId::CallRequest:
+            {
+                logger.logDebug("BtsPort, Call request from:", from);
+                handler->handleCallRequest(from);
+                break;
+            }
+            case common::MessageId::CallAccepted:
+            {
+                logger.logDebug("BtsPort, Call accepted from", from);
+                handler->handleCallAccepted(from);
+                break;
+            }
+            case common::MessageId::CallTalk:
+            {
+                logger.logDebug("BtsPort, Call talk", from);
+                handler->handleTalkMessage(reader.readRemainingText());
+                break;
+            }
+            case common::MessageId::UnknownRecipient:
+            {
+                logger.logError("Unknown recipient.");
+                common::MessageId failingMessageId = reader.readMessageId();
+                assert(failingMessageId != messageId);
```

Comment:
Nie rozumiem co ma na celu sprawdzać ten assert. Czy te id mogą być równe? Przecież BTS wiadomość UnknownRecipient przesyła z nagłówkiem starej wiadomości i to odczytujesz.